src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -1398,6 +1455,104 @@ private MqttServerStream resolvePublisher(
             return stream;
         }
 
+        private void resolveSession(
+            long traceId,
+            long authorization,
+            int flags,
+            int willDelay,
+            byte willQos,
+            byte willRetain,
+            int expiryInterval,
+            String16FW contentType,
+            MqttPayloadFormat payloadFormat,
+            String16FW responseTopic,
+            OctetsFW correlationData,
+            OctetsFW willPayload)
+        {
+            final String topic = String.format(SESSION_WILDCARD_TOPIC_FORMAT, clientId);
+            final int topicKey = topicKey(topic);
+
+            final RouteFW route = resolveTarget(routeId, authorization, topic,  PUBLISH_ONLY);
+
+            MqttSessionStateStream stream = sessionStreams.computeIfAbsent(topicKey, s -> new MqttSessionStateStream(0, topic));
+            if (route != null)
+            {
+                final long resolvedId = route.correlationId();
+                stream.initRoute(resolvedId);
+
+                final JsonObject sessionObj = objectBuilder.add(SESSION_EXPIRES_AT_NAME, sessionExpiresAt)
+                                                           .add(WILL_DELAY_NAME, willDelay)
+                                                           .build();
+
+                final OctetsFW sessionPayload = sessionPayloadRW.wrap(payloadBuffer, 0, payloadBuffer.capacity())
+                                                         .set(sessionObj.toString().getBytes(UTF_8))
+                                                         .build();
+
+                stream.doApplicationBeginOrFlush(traceId, authorization, affinity, topic, NO_FLAGS, 0, PUBLISH_AND_SUBSCRIBE);
+
+                final int payloadSize = sessionPayload.sizeof() + willPayload.sizeof();
+
```

Comment:
The logic to test if the session messages can be published will always fail when the stream has just been created, because it won't yet have any stream budget, but it looks like the intent is to re-trigger the decode process when window arrives.

Please confirm that intent, and also that there are no unwanted side effects to running this code multiple times before sufficient budget becomes available.